src/System.Security.Cryptography.Xml/tests/TransformTest.cs

```diff
@@ -57,11 +61,48 @@ public override Type[] OutputTypes
 
     public class TransformTest
     {
+        [Fact]
+        public void Constructor()
+        {
+            ConcreteTransform concreteTransform = new ConcreteTransform();
+            Assert.Null(concreteTransform.Context);
+            Assert.Null(concreteTransform.Algorithm);
+#if !NET45 && !NET451 && !NET452 && !NET46 && !NET461 && !NET462 && !NET463
+            Assert.Empty(concreteTransform.PropagatedNamespaces);
+#endif
+        }
 
         [Fact]
         public void GetDigestedOutput_Null()
         {
             Assert.Throws<NullReferenceException>(() => new ConcreteTransform().GetDigestedOutput(null));
         }
+
+#if !NET45 && !NET451 && !NET452 && !NET46 && !NET461 && !NET462 && !NET463
+        // Use a selection of transforms that do not modify the XML if applied.
+        [Theory]
+        [InlineData(typeof(XmlDsigC14NTransform))]
+        [InlineData(typeof(XmlDsigExcC14NTransform))]
+        [InlineData(typeof(XmlDsigC14NWithCommentsTransform))]
+        public void PropagatedNamespaces(Type type)
+        {
+            XmlDocument doc = new XmlDocument();
+            doc.AppendChild(doc.CreateElement("foo", "urn:foo"));
+            doc.DocumentElement.AppendChild(doc.CreateElement("bar", "urn:bar"));
+            Assert.Equal(string.Empty, doc.DocumentElement.GetAttribute("xmlns:f"));
+            Transform transform = type.GetConstructor(Type.EmptyTypes).Invoke(new object[0]) as Transform;
+            transform.LoadInput(doc);
+            transform.PropagatedNamespaces.Add("f", "urn:foo");
+            transform.PropagatedNamespaces.Add("b", "urn:bar");
```

Comment:
From what I understand, the order is unimportant. If the `References` property is not set, `PropagatedNamespaces` now returns an empty collection instead of throwing an exception. I might have geeked out trying to make the smallest change possible to make `PropagatedNamespaces` work correctly. However, it appears to work as originally intended, at least as far as I can tell.